### PR TITLE
Dpi in default theme

### LINKED ISSUE #1214: Active tab indicator square is not scaled with DPI
Output of `awesome --version`:

```
awesome v3.5.2-2174-g588cbc1 (The Fox)
 • Compiled against Lua 5.3.3 (running with Lua 5.3)
 • D-Bus support: ✔
 • execinfo support: ✔
 • RandR 1.5 support: ✔
 • LGI version: 0.9.1
```

**How to reproduce the issue:**

1. Download [`awesome-git`](https://aur.archlinux.org/packages/awesome-git)
1. Unpack it
1. Apply the following diff to `PKGBUILD`:

        20c20
        <          'lua' 'lua-lgi-git' 'pango' 'startup-notification' 'xcb-util-cursor'
        ---
        >          'lua' 'lua-lgi' 'pango' 'startup-notification' 'xcb-util-cursor'
        22c22
        < makedepends=('asciidoc' 'cmake' 'docbook-xsl' 'doxygen' 'imagemagick' 'ldoc-git' 'xmlto')
        ---
        > makedepends=('asciidoc' 'cmake' 'docbook-xsl' 'doxygen' 'imagemagick' 'xmlto')
        47c47,48
        <     -DLUA_LIBRARY=/usr/lib/liblua.so.5.3
        ---
        >     -DLUA_LIBRARY=/usr/lib/liblua.so.5.3 \
        >     -DGENERATE_DOC=False
1. Compile and install using `makepkg --syncdeps --install`
1. Set up your X resources:

        $ cat /etc/X11/Xresources
        Xft.dpi:        276
1. Restart awesome

**Actual result:**

The tab bar font scales up, but the tab state indicator square is still tiny: 2016-11-09-202242_810x71_scrot

**Expected result:**

The tab state indicator square should scale with DPI. Possibly it could be a glyph rather than an image?

### PR DESCRIPTION
**Pull request by actionless**
closes #1214

### CODE REVIEW

File: themes/default/theme.lua [hunk 1]
```diff
@@ -2,6 +2,11 @@
 -- Default awesome theme --
 ---------------------------
 
+local beautiful = require("beautiful")
+local xresources = beautiful.xresources
+local dpi = xresources.apply_dpi
+local theme_assets = beautiful.theme_assets
```
Comment (psychon) [inline themes/default/theme.lua]: Isn't this another cyclic dependency waiting to become a problem? It's no problem right now, but can become one in the future (beautiful depends on theme which depends on beautiful).

I'd propose not to `require("beautiful")` here, but directly the needed parts.

File: themes/sky/theme.lua [hunk 2]
```diff
@@ -47,15 +53,22 @@ theme.awesome_icon           = "@AWESOME_THEMES_PATH@/sky/awesome-icon.png"
...
+    taglist_square_size, theme.fg_normal
+)
 
 -- MISC
 theme.wallpaper             = "@AWESOME_THEMES_PATH@/sky/sky-background.png"
 theme.taglist_squares       = "true"
 theme.titlebar_close_button = "true"
-theme.menu_height           = 15
-theme.menu_width            = 100
+theme.menu_height = dpi(16)
```
Comment (psychon) [inline themes/sky/theme.lua]: Is the change 15 -> 16 intended?

File: themes/sky/theme.lua [hunk 3]
```diff
@@ -4,6 +4,12 @@
 -------------------------------
 -- If you want SVGs and extras, get them from garoth.com/awesome/sky-theme
 
+local beautiful = require("beautiful")
+local xresources = beautiful.xresources
+local xrdb = xresources.get_current_theme()
```
Comment (psychon) [inline themes/sky/theme.lua]: Why xrdb?

File: themes/sky/theme.lua (hunk 2 above)
Comment (actionless) [inline themes/sky/theme.lua]: nope, by mistake

**Comment by actionless** [comment]
@psychon fixed

**Comment by actionless** [comment]
done

**Comment by psychon** [comment]
Now LDoc is unhappy:
```
/home/travis/build/awesomeWM/awesome/.build-testing-gce-702733d5-ff7b-4572-b0b3-9bf87a234d0d-x86_64-linux-gnu-4.8/lib/beautiful/theme_assets.lua:1: contains no items
```

**Comment by actionless** [comment]
any ideas how this travis failure can be related to these changes?
https://travis-ci.org/awesomeWM/awesome/jobs/185041642

**Comment by Elv13** [comment]
"awesome" is normally a global variable. If it isn't set, then something overwrote it with `nil`, but nothing in your patchset does that. Did you rebase this correctly on top of the master branch?

**Comment by psychon** [comment]
`beautiful` now requires `beautiful.theme_assets` which it previously did not do. `theme_assets` require()s `gears` and so `gears.timer` is loaded while it previously was not loaded. The unit tests run by `busted` are supposed to be *unit*  tests and so do not mock the whole API. Specifically, there is no (and never was) `awesome` global available.

Simple work around would be to change `assets.lua` to `require("gears.color")` instead of all of `gears`. It doesn't use anything but `gears.color` any way.

**Comment by actionless** [comment]
@psychon 
oh, right, i was fixing that already in the original commits but lost it accidentally while resolving merge conflict, i will update it later on

**Comment by actionless** [comment]
updated and rebased (again :) )

let's wait for travis

**Comment by blueyed** [comment]
@actionless 
Tests/CI checks every commit by now (to help with git-bisect), and currently it fails for the following two still: 3053669b07b11efc30d37973366e694502f54720 4f64a00774c054546e76dc00983a4a3c6f49dd06.
You should probably rebase it to squash the fix, or move it before those commits.

**Comment by actionless** [comment]
oh, my bad, squashed

**Comment by actionless** [comment]
:tada:

**Comment by Elv13** [comment]
Just a little question before merging. Should them_asset be a private API instead of a public one? We can't just break its API as it would break people themes, but still.

**Comment by psychon** [comment]
What would you count as "private API"? An undocumented one?
That doesn't really work since people will copy the themes we ship and do modifications, so it cannot just break.
Or are you referring to "this commit changes API" by breaking "old" versions of the `xresources` theme?

**Comment by psychon** [comment]
> will have to wait for the v5 API break to be merged

I know you want to avoid a "4 years of development on master", but "collecting" pull requests until we decide its time to break the API is not going to fly either. We can either have a "v4" branch and a "master" branch and try to backport everything to the "v4" branch or we can (at least initially, might want to switch to a different strategy later) have a "v5" branch for API breaks and continue development on master (and then regularly (after releases?)) merge master into v5).

Anyway, thanks for catching this API break.

> This time I guess @actionless can add some compatibility alias to avoid breaking the existing themes. 

So, we need a file `themes/xresources/assets.lua` containing something like the following:
```lua
require("awful.util").deprecate("foo")
return require("beautiful.xresources")
```
Right?

**Comment by Elv13** [comment]
> I know you want to avoid a "4 years of development on master", but "collecting" pull requests until we decide its time to break the API is not going to fly either. We can either have a "v4" branch and a "master" branch and try to backport everything to the "v4" branch or we can (at least initially, might want to switch to a different strategy later) have a "v5" branch for API breaks and continue development on master (and then regularly (after releases?)) merge master into v5).

This isn't going to work. This is what we did last time. Doing the opposite (working on master and having a v5 branch) is already a better option. But just working harder to keep the API compatible, even if it requires dummy files, FIXME and an improved deprecation framework requires *less* work than dealing with 2 branches. 95% of the breakage can either be avoided or handled transparently.

**Comment by actionless** [comment]
hopefully this time everything will be fine

**Comment by actionless** [comment]
@Elv13 , what do you think?

**Comment by blueyed** [comment]
Thanks!

**Pull request merged**